plugins/sentinelone/komand_sentinelone/actions/quarantine_multiple/action.py

```diff
@@ -0,0 +1,55 @@
+import insightconnect_plugin_runtime
+from .schema import QuarantineMultipleInput, QuarantineMultipleOutput, Input, Output, Component
+
+# Custom imports below
+from insightconnect_plugin_runtime.exceptions import PluginException
+from requests.exceptions import HTTPError
+from komand_sentinelone.util.helper import Helper
+
+
+class QuarantineMultiple(insightconnect_plugin_runtime.Action):
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="quarantine_multiple",
+            description=Component.DESCRIPTION,
+            input=QuarantineMultipleInput(),
+            output=QuarantineMultipleOutput(),
+        )
+
+    def run(self, params={}):
+        # START INPUT BINDING - DO NOT REMOVE - ANY INPUTS BELOW WILL UPDATE WITH YOUR PLUGIN SPEC AFTER REGENERATION
+        agents = list(set(params.get(Input.AGENTS, [])))
+        whitelist = params.get(Input.WHITELIST, None)
+        quarantine_state = params.get(Input.QUARANTINE_STATE)
+        # END INPUT BINDING - DO NOT REMOVE
+
+        successful = []
+        failures = []
+        for agent in agents:
+            agents_founds = self.connection.client.search_agents(agent, results_length=2)
+
+            if not Helper.check_agents_found(agents_founds):
+                error = f"No agents found using the host information: {agent}."
+                self.logger.info(error)
+                failures.append(Helper.return_failure_details(agent, error))
+                continue
+
+            agent_obj = agents_founds[0]
+            payload = {"ids": [agent_obj["id"]]}
+
+            try:
+                action_type = "connect"
+                if quarantine_state:
+                    if Helper.check_disconnected(agent_obj):
+                        self.logger.info(f"Agent: {agent} is already quarantined")
+                    if whitelist:
+                        Helper.find_in_whitelist(agent_obj, whitelist)
+                    action_type = "disconnect"
+                self.connection.agents_action(action_type, payload)
+                successful.append(agent)
+            except (PluginException, HTTPError) as error:
```

Comment:
I think this is a good way to handle agents who are already found quarantined. May wish to check with Connisoars on whether this is expected response however. As in, should they be added to success also?

Reply:
Hi 😄 Talked with them and thats fine to append `completed` array with agents who are already found quarantined 😄